mongo/integration/client_test.go

```diff
@@ -597,6 +598,63 @@ func TestClient(t *testing.T) {
 		closed := len(tpm.Events(func(e *event.PoolEvent) bool { return e.Type == event.ConnectionClosed }))
 		assert.Equal(t, 0, closed, "expected no connections to be closed")
 	})
+
+	// Test that OP_MSG is used for authentication-related commands on 3.6+ (WV 6+). Do not test when API version is
+	// set, as handshakes will always use OP_MSG.
+	opMsgOpts := mtest.NewOptions().ClientType(mtest.Proxy).MinServerVersion("3.6").Auth(true).RequireAPIVersion(false)
+	mt.RunOpts("OP_MSG used for authentication on 3.6+", opMsgOpts, func(mt *mtest.T) {
+		err := mt.Client.Ping(context.Background(), mtest.PrimaryRp)
+		assert.Nil(mt, err, "Ping error: %v", err)
+
+		msgPairs := mt.GetProxiedMessages()
+		assert.True(mt, len(msgPairs) >= 2, "expected at least 2 events, got %v", len(msgPairs))
+
+		// First two messages should be connection handshakes: one for the heartbeat connection and the other for the
+		// application connection. These handshakes should use OP_QUERY as their OpCode, as wire version is not yet
+		// known.
```

Comment:
The RTT monitor connection is also created in [Server#Connect here](https://github.com/mongodb/mongo-go-driver/blob/master/x/mongo/driver/topology/server.go#L213).

I think the RTT monitor connection handshake may occur any time in this sequence.

I think this assertion is valuable to ensure we are speaking OP_QUERY on the first message (and not inadvertently OP_MSG). I suggest only checking the first message, and updating the comment as follows:

```suggestion
		// First messages should be connection handshakes. The handshake should use OP_QUERY as the OpCode, as
		// wire version is not yet known.
```